app/slave/cluster_slave.py

```diff
@@ -74,24 +74,27 @@ def _start_heartbeat(self):
 
     def _run_heartbeat(self):
         try:
-            self._send_heartbeat_to_master()
-            self._heartbeat_failure_count = 0
+            response = self._send_heartbeat_to_master()
+
+            # A non HTTP 200 response indicates that the master is up, but does not recognize the slave.
+            if not response.ok:
+                self._logger.error('Master did not respond with HTTP 200')
```

Comment:
Do you want to log what the actual response code was here? Might be useful for debugging.